monai/apps/reconstruction/networks/nets/utils.py

```diff
@@ -0,0 +1,182 @@
+
+    Args:
+        x: input of shape (B,C*2,H,W) for 2D data or (B,C*2,H,W,D) for 3D data
+
+    Returns:
+        output of shape (B,C,H,W,2) for 2D data or (B,C,H,W,D,2) for 3D data
+    """
+    if x.shape[1] % 2 != 0:
+        raise ValueError(f"channel dimension should be even but ({x.shape[1]}) is odd.")
+
+    if len(x.shape) == 4:  # this is 2D
+        b, c2, h, w = x.shape  # c2 means c*2
+        c = c2 // 2
+        return x.view(b, 2, c, h, w).permute(0, 2, 3, 4, 1)
+
+    elif len(x.shape) == 5:  # this is 3D
+        b, c2, h, w, d = x.shape  # c2 means c*2
+        c = c2 // 2
+        return x.view(b, 2, c, h, w, d).permute(0, 2, 3, 4, 5, 1)
+
+
+def complex_normalize(self, x: Tensor) -> Sequence:
+    """
+    Performs group mean-std normalization for complex data. Normalization is done for each batch member
+    and each part (real and imaginary parts), separately. To see what "group" means, mean of
+    an input of shape (B,C,H,W) will be (B,).
+
+    Args:
+        x: input of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+
+    Returns:
+        A tuple containing
+            (1) normalized output of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+            (2) mean
+            (3) std
+    """
+    if len(x.shape) == 4:  # this is 2D
+        b, c, h, w = x.shape
+        x = x.contiguous().view(b, 2, c // 2 * h * w)
+        mean = x.mean(dim=2).view(b, 2, 1, 1, 1).expand(b, 2, c // 2, 1, 1).contiguous().view(b, c, 1, 1)
+        std = x.std(dim=2, unbiased=False).view(b, 2, 1, 1, 1).expand(b, 2, c // 2, 1, 1).contiguous().view(b, c, 1, 1)
+        x = x.view(b, c, h, w)
+        return (x - mean) / std, mean, std
+
+    elif len(x.shape) == 5:  # this is 3D
+        b, c, h, w, d = x.shape
+        x = x.contiguous().view(b, 2, c // 2 * h * w * d)
+        mean = x.mean(dim=2).view(b, 2, 1, 1, 1, 1).expand(b, 2, c // 2, 1, 1, 1).contiguous().view(b, c, 1, 1, 1)
+        std = x.std(dim=2, unbiased=False).view(b, 2, 1, 1, 1, 1).expand(b, 2, c // 2, 1, 1, 1).contiguous().view(b, c, 1, 1, 1)
+        x = x.view(b, c, h, w, d)
+        return (x - mean) / std, mean, std
+
+
+def reverse_complex_normalize(self, x: Tensor, mean: float, std: float) -> Tensor:
+    """
+    Reverses the normalization done by complex_normalize
+
+    Args:
+        x: input of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+        mean: mean before normalization
+        std: std before normalization
+
+    Returns:
+        denormalized output of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+    """
+    return x * std + mean
+
+
+def floor_ceil(n: float) -> Sequence:
+    """
+    Returns floor and ceil of the input
+
+    Args:
+        n: input number
+
+    Returns:
+        A tuple containing:
+            (1) floor(n)
+            (2) ceil(n)
+    """
+    return math.floor(n), math.ceil(n)
+
+
+def pad(self, x: Tensor) -> Sequence:
+    """
```

Comment:
I was wondering iif the goal is to perform `DivisiblePad`. Could help check it? https://github.com/Project-MONAI/MONAI/blob/46d9026c1ad5182ea8edc780771eb31929947b7c/monai/transforms/croppad/array.py#L322

Thanks!

Reply:
Now I got it. It is not `DivisiblePad`. Would it be `SpatialPad`?

Reply:
Yeah could be, I'll check whether SpatialPad can do this.